From 2e96a014f258cc5297bf5f026646666cf65e2398 Mon Sep 17 00:00:00 2001
From: Takahiro Ueda <t.ueda.od@juntendo.ac.jp>
Date: Mon, 23 Mar 2026 22:02:28 +0900
Subject: [PATCH] feat: allow skipping HEPToolsInstallers refresh

Skip refreshing HEPToolsInstallers when
MG5_NO_REFRESH_HEPTOOLS_INSTALLERS is set.
---
 madgraph/interface/madgraph_interface.py | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/madgraph/interface/madgraph_interface.py b/madgraph/interface/madgraph_interface.py
index 24ebe9701..45535c2c9 100755
--- a/madgraph/interface/madgraph_interface.py
+++ b/madgraph/interface/madgraph_interface.py
@@ -6068,7 +6068,10 @@ This implies that with decay chains:
         add_options = list(additional_options)
 
         # Always refresh the installer if already present
-        if not os.path.isdir(pjoin(MG5DIR,'HEPTools','HEPToolsInstallers')):
+        # unless MG5_NO_REFRESH_HEPTOOLS_INSTALLERS is set
+        no_refresh_heptools_installers = os.environ.get('MG5_NO_REFRESH_HEPTOOLS_INSTALLERS','')
+        has_heptools_installers = os.path.isdir(pjoin(MG5DIR,'HEPTools','HEPToolsInstallers'))
+        if not has_heptools_installers:
             if HepToolsInstaller_web_address is None:
                 raise MadGraph5Error("The option 'HepToolsInstaller_web_address'"+\
                              " must be specified in function advanced_install"+\
@@ -6076,10 +6079,13 @@ This implies that with decay chains:
             if not os.path.isdir(pjoin(MG5DIR,'HEPTools')):
                 os.mkdir(pjoin(MG5DIR,'HEPTools'))
         elif not HepToolsInstaller_web_address is None:
-            shutil.rmtree(pjoin(MG5DIR,'HEPTools','HEPToolsInstallers'))
+            if not no_refresh_heptools_installers:
+                shutil.rmtree(pjoin(MG5DIR,'HEPTools','HEPToolsInstallers'))
         if not HepToolsInstaller_web_address is None:
+            if has_heptools_installers and no_refresh_heptools_installers:
+                pass
             # Download HEPToolsInstaller (unless the --local option is used)
-            if not '--local' in add_options:
+            elif not '--local' in add_options:
                 logger.info('Downloading the HEPToolInstaller at:\n   %s'%
                                                       HepToolsInstaller_web_address)
                 # Guess if it is a local or web address
-- 
2.34.1

